### PR TITLE
Scripts/Ebon Hold: Into the Realm of Shadows

### PR DESCRIPTION
**Pull request by Sorikoff**
**Changes proposed:** 
- Removed obsolete C++ script.
- Created separate SAI scripts for `Dark Rider of Acherus`, `Acherus Deathcharger`, and `Salanar the Horseman`.
- Changed spell click spells to correct ones.
- Slightly reorganized code.

**Target branch(es):** 3.3.5

~**Issues addressed:**~

**Tests performed:** Works as intended.

**Known issues and TODO list:**:
- [ ] after removing player from vehicle, camera is getting stuck in place.
- [ ] personal spawn support in core.

### CODE REVIEW

**Comment by illfated** [comment]
Cosmetic/codestyle: missing NewLine at the end of [9999_99_99_99_world.sql](https://github.com/TrinityCore/TrinityCore/pull/25617/files#diff-41736fd024e478d43ea196b36e4c48290630e19da51c8bb8cff3b705199330e6) .

File: sql/updates/world/3.3.5/9999_99_99_99_world.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+DELETE FROM `npc_spellclick_spells` WHERE `npc_entry`=28782;
+INSERT INTO `npc_spellclick_spells` (`npc_entry`,`spell_id`,`cast_flags`,`user_type`) VALUES
+(28782,46598,1,0),
+(28782,52349,1,0);
+
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=18 AND `SourceGroup`=28782;
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceEntry`=52349;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
+(18,28782,46598,0,0,31,0,3,0,0,0,0,0,'','Allow spellclick for NPC'),
+(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'','Required quest 12687 active for spellclick'),
```
Comment (illfated) [inline sql/updates/world/3.3.5/9999_99_99_99_world.sql:37]: Suggestion for improvement of the `Comment` field:
```suggestion
(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'',"Requires quest 'Into the Realm of Shadows' active for spellclick"),
```

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 2]
```diff
@@ -865,87 +824,21 @@ class spell_stable_master_repo : public AuraScript
...
-        Unit* target = GetHitUnit();
-        target->RemoveAurasDueToSpell(SPELL_EFFECT_STOLEN_HORSE);
-
-        Unit* caster = GetCaster();
-        caster->RemoveAurasDueToSpell(SPELL_EFFECT_STOLEN_HORSE);
-        caster->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_SPELLCLICK);
-        caster->SetFaction(FACTION_FRIENDLY);
-
-        caster->CastSpell(caster, SPELL_CALL_DARK_RIDER, true);
+        GetHitUnit()->ExitVehicle();
```
Comment (Sorikoff) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:833]: After removing aura `52349` camera is getting stuck in place (character moves though). Could someone please help with this issue? Currently have no idea.

Comment (illfated) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:833]: Maybe I have misunderstood something in these changes, but while viewing the code in split screen, it looks like aura 52349 is not handled by any enum or reference. Could that be part of the issue?

Comment (Sorikoff) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:833]: It's not used in C++ code (only applied in SQL).

Comment (illfated) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:833]: Fair enough, I misunderstood how it is handled.

**Comment by Riztazz** [comment]
This quest is supposed to use a spell to summon the final NPC, 52363, on top of that, that spawn is a personal spawn (i don't think 3.3.5 has that implemented but pretty sure master does) and from what we've gathered with @Traesh a year ago, it's a personal spawn.
https://github.com/TrinityCore/TrinityCore/blob/c250a858b91a11736ee0a64c2f120b1d036b5203/src/server/game/DataStores/DBCEnums.h#L1346

**Comment by Sorikoff** [comment]
Thank you for this note! As far as I see, this indeed is not implement in 3.3.5. https://github.com/TrinityCore/TrinityCore/commit/b7bb5e6a98119512702519adecf559ecedc20743 probably needs to be ported.

File: sql/updates/world/3.3.5/9999_99_99_99_world.sql [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+DELETE FROM `npc_spellclick_spells` WHERE `npc_entry`=28782;
+INSERT INTO `npc_spellclick_spells` (`npc_entry`,`spell_id`,`cast_flags`,`user_type`) VALUES
+(28782,46598,1,0),
+(28782,52349,1,0);
+
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=18 AND `SourceGroup`=28782;
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceEntry`=52349;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
+(18,28782,46598,0,0,31,0,3,0,0,0,0,0,'','Allow spellclick for NPC'),
+(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'','Requires quest 'Into the Realm of Shadows' active for spellclick'),
```
Comment (illfated) [inline sql/updates/world/3.3.5/9999_99_99_99_world.sql:50]: To use a double set of quotes in a comment, you need to use 1 set of double quotes and 1 set or single quotes, or use an escape character (backslash) for the inner single quotes. My suggestion:
```suggestion
(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'',"Requires quest 'Into the Realm of Shadows' active for spellclick"),
```
(CircleCI codestyle_and_sql https://app.circleci.com/pipelines/github/TrinityCore/TrinityCore/2992/workflows/70193d78-f2a1-467d-963b-33db6e28192e/jobs/12641)

**Comment by Ovahlord** [comment]
You forgot to remove the scriptname for npc_salanar_the_horseman and you did not add the SAI for him yet so you cannot enter the shadow realm, Jimbo.

However, I went on testing this stuff and so far the Horse neither the rider do reset property so they are stuck in combat. Another thing that I have noticed is that Salanar does not walk towards you when delivering the horse which he should do when you call him. About the vehicle camera issue, I was not able to reproduce it after using.

```
        if (Vehicle* vehicle = GetHitUnit()->GetVehicle())
            vehicle->RemoveAllPassengers();
```

**Comment by Sorikoff** [comment]
Thank you for feedback. Removed ScriptName. Not sure about SAI for him though as you enter shadow realm automatically. At least it works that way for me.

Vehicle camera issue is still present for me. Will check reset.

File: sql/updates/world/3.3.5/9999_99_99_99_world.sql [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
```
Comment (illfated) [inline sql/updates/world/3.3.5/9999_99_99_99_world.sql:52]: Looks like you removed the solution to use only 1 pair each of double quotes (" ") and single quotes (' ').

Maybe it is easier for you to remember if you use the escape character "backslash" \ for the inner quote pair, which is actually meant to transfer into the DB comment text field.

```suggestion
(18,28782,52349,0,0,9,0,12687,0,0,0,0,0,'','Requires quest \'Into the Realm of Shadows\' active for spellclick'),
```

Comment (Sorikoff) [reply sql/updates/world/3.3.5/9999_99_99_99_world.sql:52]: Yeah, I just used force push, and forgot to pull PR changes.

**Comment by jackpoz** [comment]
> * personal spawn support in core.

this has been added with https://github.com/TrinityCore/TrinityCore/commit/3b2c878dd067f49ee48ddfc27c62eb4e1fe619d9 , please update this PR to include whatever required personal spawns

**Comment by Faq** [comment]
@Sorikoff You will open new PR?

**Comment by illfated** [comment]
Looks like `https://github.com/Sorikoff` does not contain any current repository. Will have to re-fork TC first to start working on it from that account.

**Comment by jackpoz** [comment]
a new PR with just these commits should be re-opened:
1. https://github.com/TrinityCore/TrinityCore/pull/25617/commits/9895ab9104113af633f535e4957c30537720dfdd
2. https://github.com/TrinityCore/TrinityCore/pull/25617/commits/7b160b3bd2bfe6c38c26820d27a3375e980cdc53
3. https://github.com/TrinityCore/TrinityCore/pull/25617/commits/73d1b35adad543d3ecfffced0f46bbbcf9ebb055
4. https://github.com/TrinityCore/TrinityCore/pull/25617/commits/3f3d45cd3f429d8d25da344bde0cc71d08d8312c
5. https://github.com/TrinityCore/TrinityCore/pull/25617/commits/68f525559c1f60f0df59cef9425210773e9789ad

**Comment by illfated** [comment]
I presume it will be useful for me to open a new PR just to consolidate those commits into 1, in case anyone else wants to work on adding any missing C++ coding or SAI. (Also useful to have it open and available for review, suggestions etc.)

**Comment by jackpoz** [comment]
@illfated if possible please keep the 5 commits split in the PR

**Comment by illfated** [comment]
> @illfated if possible please keep the 5 commits split in the PR

Very well. Thanks for the correction. Glad I have not started doing anything major, only mixed together the SQL part, so it is easy enough to use the commits. I presume any updates (improvements) from my side will have to be separate commits as well.

**Comment by offl** [comment]
> in case anyone else wants to work on adding any missing C++ coding or SAI. (Also useful to have it open and available for review, suggestions etc.)

This PR was opened for 2 months for review and suggestions
Merging PR mostly requires changes from OP and not from reviewers

**Comment by illfated** [comment]
Sure, that is why I offered to open a new PR for the changes in here (with a note saying that @Sorikoff could have the credit, or even remake this original PR).

**Comment by offl** [comment]
I mean if you are not going to update and test it, very likely it will be too not merged

**Comment by illfated** [comment]
Fair point. I thought as much, but also thought that it would be useful to have the material in an open PR. Oh well. Point taken.

**Pull request closed without merging**